aws/resource_aws_iam_role.go

```diff
@@ -110,6 +110,36 @@ func resourceAwsIamRole() *schema.Resource {
 			},
 
 			"tags": tagsSchema(),
+
+			"inline_policy": {
+				Type:       schema.TypeSet,
+				Optional:   true,
+				Computed:   true,
+				ConfigMode: schema.SchemaConfigModeAttr,
```

Comment:
We should remove the legacy attribute mode from this block:

```suggestion
				Type:     schema.TypeSet,
				Optional: true,
				Computed: true,
```

While it can potentially help solve the "configure this block to 0 items" problem, it creates quite a few other non-obvious practitioner problems and is generally not meant to be supported for new attributes. These include:

- Confusing "attribute is required" errors for any future `Optional` nested attributes in attribute mode
- Removal of support for block mode in JSON syntax

The current design of Terraform (both sides of the protocol barrier) is not able to cleanly resolve all situations where practitioners may want to exclusively manage a particular block configuration. The next version of the Terraform Plugin Protocol and future work in the Terraform Plugin SDK(/Go) is being designed around solving these particular issues. 

References:

- https://github.com/hashicorp/terraform/issues/27745#issuecomment-777623731
- https://www.terraform.io/docs/language/attr-as-blocks.html#in-json-syntax
- https://github.com/hashicorp/terraform-provider-aws/issues/17618
- https://github.com/hashicorp/terraform-provider-aws/issues/17356
- https://github.com/hashicorp/terraform-provider-aws/issues/13854
- https://github.com/hashicorp/terraform-provider-aws/issues/13003
- https://github.com/hashicorp/terraform-provider-aws/issues/9201
- https://github.com/hashicorp/terraform-provider-aws/issues/8786
- https://github.com/hashicorp/terraform-provider-aws/pull/8187
- https://github.com/hashicorp/terraform/pull/27699

Reply:
Done